src/consensus/pbft/libbyz/Replica.cpp

```diff
@@ -164,11 +168,26 @@ Replica::Replica(
   exec_command = nullptr;
   non_det_choices = 0;
 
-  if (ledger)
-  {
-    ledger_replay = std::make_unique<LedgerReplay>();
-    ledger_writer = std::make_unique<LedgerWriter>(std::move(ledger));
-  }
+  ledger_writer = std::make_unique<LedgerWriter>(store, pbft_pre_prepares);
+  requests_local_hook = [this](
+                          kv::Version version,
+                          const pbft::Requests::State& s,
+                          const pbft::Requests::Write& w) {
+    for (auto& [key, value] : w)
+    {
+      playback_request(value.value);
+    }
+  };
```

Comment:
Can this be just a function in the replica class?

Reply:
As in not be a local hook on a map? How will we know to call the functions after we call `Store::deserialise()`?

Reply:
Do not create a hook here, Have a function that takes the same input and then loops over it an calls playback_request. Then when you set the hooks create the std::functions there. The std::functions do not need to be stored in replica object